Trivial changes to AMOEBA plugin.

andysim opened the pull request:
# Description

Just a couple of small changes to the AMOEBA code, to make the CUDA tests pass cleanly, and to make the code more readable.

# Status
- [x] Rename some variables, to avoid the confusion raised in #1688.
- [x] Avoid finite difference test in single precision, which fails due to roundoff error.
- [x] Ready to go!

peastman commented:
What test was failing?  I haven't had any problems with that function.  The step size is big enough that it ought to work in single precision.

andysim commented:
The Cuda AMOEBA extrapolated test's finite difference test was failing (by a tiny amount) on our K20 and K40 cards.  I'm not sure why it works for you, but feel free to ignore this change is you think it's not needed. 

> On Jan 19, 2017, at 6:32 PM, peastman <notifications@github.com> wrote:
> 
> What test was failing? I haven't had any problems with that function. The step size is big enough that it ought to work in single precision.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

peastman commented:
If that's the case, how about just increasing the tolerance slightly?  In the last line of `check_finite_differences()`, try changing the 1e-4 to 2e-4.  Does that make it pass?  We could even make it use a higher tolerance in double precision.  Something like:

```C++
double tol = 2e-4;
if (Platform::getPlatformByName("CUDA").getPropertyValue(context, "Precision") == "double")
    tol = 1e-6;
```

andysim commented:
I did a little testing, and found that the step size is pretty good:

```
1e-4: Expected 379.946, found 379.665
5e-4: Expected 379.946, found 379.972
1e-3: Expected 379.946, found 379.957
5e-3: Expected 379.946, found 379.984
1e-2: Expected 379.946, found 380.069 
```

It looks like the precision argument was being ignored previously so I fixed that, and implemented the tolerance check as you suggested.  Again, feel free to merge / ignore / squash this PR as you see fit.

peastman commented:
Thanks!  This looks good now.  Since we're just about to build the release candidate for 7.1, I'll wait to merge this until after that is released.
